src/OneWayBondingCurve.sol

```diff
@@ -87,49 +79,75 @@ contract OneWayBondingCurve {
-        return FixedPointMathLib.mulDivDown(amount, USDC_BASE, BAL_BASE);
+    /// @notice Deposit remaining USDC in Aave V2 Collector after 100k BAL Amount Cap has been filled
+    function depositUsdcCollector() external {
+        uint256 usdcBalance = USDC.balanceOf(AaveV2Ethereum.COLLECTOR);
+        uint256 usdcAllowance = USDC.allowance(AaveV2Ethereum.COLLECTOR, address(this));
+
+        if (totalBalReceived < BAL_AMOUNT_CAP) revert BalCapNotFilled();
+        if (usdcAllowance == 0) revert ZeroAllowance();
+        if (usdcBalance == 0) revert ZeroBalance();
+
+        // USDC available to Bonding Curve to spend on behalf of Aave V2 Collector
+        uint256 usdcAmount = (usdcAllowance <= usdcBalance) ? usdcAllowance : usdcBalance;
+
+        USDC.safeTransferFrom(AaveV2Ethereum.COLLECTOR, address(this), usdcAmount);
+        USDC.approve(address(AaveV2Ethereum.POOL), usdcAmount);
+        AaveV2Ethereum.POOL.deposit(address(USDC), usdcAmount, AaveV2Ethereum.COLLECTOR, 0);
+
+        emit Deposit(address(USDC), address(AUSDC), usdcAmount);
     }
 
-    /// @notice Normalize BAL/USD Chainlink Oracle Feed decimals to USDC decimals (6)
-    function normalizeFromOracleDecimalstoUSDCDecimals(uint256 amount) public view returns (uint256) {
-        return FixedPointMathLib.mulDivDown(amount, USDC_BASE, 10**uint256(BAL_USD_FEED.decimals()));
+    /// @notice Deposit all acquired BAL after 100k BAL Amount Cap has been filled
+    function depositBalCollector() external {
+        uint256 balBalance = BAL.balanceOf(AaveV2Ethereum.COLLECTOR);
+        uint256 balAllowance = BAL.allowance(AaveV2Ethereum.COLLECTOR, address(this));
+
+        if (totalBalReceived < BAL_AMOUNT_CAP) revert BalCapNotFilled();
+        if (balAllowance == 0) revert ZeroAllowance();
+        if (balBalance == 0) revert ZeroBalance();
+
+        // BAL available to Bonding Curve to spend on behalf of Aave V2 Collector
+        uint256 balAmount = (balAllowance <= balBalance) ? balAllowance : balBalance;
+
+        BAL.safeTransferFrom(AaveV2Ethereum.COLLECTOR, address(this), balAmount);
+        BAL.approve(address(AaveV2Ethereum.POOL), balAmount);
+        AaveV2Ethereum.POOL.deposit(address(BAL), balAmount, AaveV2Ethereum.COLLECTOR, 0);
+
+        emit Deposit(address(BAL), address(ABAL), balAmount);
     }
 
-    /// @notice The bonding curve price multiplier with arbitrage incentive
-    function getBondingCurvePriceMultiplier() public pure returns (uint256) {
-        return
-            FixedPointMathLib.mulDivDown(
-                BASIS_POINTS_GRANULARITY + BASIS_POINTS_ARBITRAGE_INCENTIVE,
-                USDC_BASE,
-                BASIS_POINTS_GRANULARITY
-            );
+    /// @notice Transfer any tokens accidentally sent to this contract to Aave V2 Collector
+    /// @param tokens List of token addresses
+    function rescueTokens(address[] memory tokens) external {
```

Comment:
why can't this be `address[] calldata tokens`

Reply:
Oh good question. I didn't think of it. Let me try it out.